src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -1706,7 +1706,6 @@ void ClassFileParser::parse_fields(const ClassFileStream* const cfs,
 
     // Remember how many oops we encountered and compute allocation type
     const FieldAllocationType atype = fac->update(is_static, type);
```

Comment:
The returned `FieldAllocationType` is never used at either call-site, so maybe the `update` method can be simplified, too? (It seems all `update` does is increment a per-type counter, so the name is a bit surprising)

Reply:
Right, I removed the value returned by the `update()` method.